src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerOptionService.cs

```diff
@@ -0,0 +1,685 @@
+    using System.Diagnostics;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Globalization;
+    using System.Security.Permissions;
+
+    /// <summary>
+    ///     Provides access to get and set option values for a designer.
+    /// </summary>
+    [HostProtection(SharedState = true)]
+    public abstract class DesignerOptionService : IDesignerOptionService
+    {
+        private DesignerOptionCollection _options;
+
+        /// <summary>
+        ///     Returns the options collection for this service.  There is 
+        ///     always a global options collection that contains child collections.
+        /// </summary>
+        public DesignerOptionCollection Options
+        {
+            get
+            {
+                if (_options == null)
+                {
+                    _options = new DesignerOptionCollection(this, null, string.Empty, null);
+                }
+                return _options;
+            }
+        }
+
+        /// <summary>
+        ///     Creates a new DesignerOptionCollection with the given name, and adds it to 
+        ///     the given parent.  The "value" parameter specifies an object whose public 
+        ///     properties will be used in the Propeties collection of the option collection.  
+        ///     The value parameter can be null if this options collection does not offer 
+        ///     any properties.  Properties will be wrapped in such a way that passing 
+        ///     anything into the component parameter of the property descriptor will be 
+        ///     ignored and the value object will be substituted.
+        /// </summary>
+        protected DesignerOptionCollection CreateOptionCollection(DesignerOptionCollection parent, string name, object value)
+        {
+            if (parent == null)
+            {
+                throw new ArgumentNullException("parent");
+            }
+
+            if (name == null)
+            {
+                throw new ArgumentNullException("name");
+            }
+
+            if (name.Length == 0)
+            {
+                throw new ArgumentException(SR.Format(SR.InvalidArgument, name.Length.ToString(CultureInfo.CurrentCulture), (0).ToString(CultureInfo.CurrentCulture)), "name.Length");
+            }
+
+            return new DesignerOptionCollection(this, parent, name, value);
+        }
+
+        /// <summary>
+        ///     Retrieves the property descriptor for the given page / value name.  Returns
+        ///     null if the property couldn't be found.
+        /// </summary>
+        private PropertyDescriptor GetOptionProperty(string pageName, string valueName)
+        {
+            if (pageName == null)
+            {
+                throw new ArgumentNullException("pageName");
+            }
+
+            if (valueName == null)
+            {
+                throw new ArgumentNullException("valueName");
+            }
+
+            string[] optionNames = pageName.Split(new char[] { '\\' });
```

Comment:
We can avoid this char array each time by caching it in a static readonly field.